Core/Core/InstUI/Views/TextSectionView.swift

```diff
@@ -20,51 +20,139 @@ import SwiftUI
 
 public extension InstUI {
     struct TextSectionView: View {
+        public struct SectionData: Identifiable, Equatable {
+            public var id: String { title + description }
+
+            public let title: String
+            public let description: String
+            public let isRichContent: Bool
+
+            public init(title: String, description: String, isRichContent: Bool = false) {
+                self.title = title
+                self.description = description
+                self.isRichContent = isRichContent
+            }
+        }
         @Environment(\.sizeCategory) private var sizeCategory
-        private let title: String
-        private let description: String
+        private let sectionData: [SectionData]
+
+        public init(
+            title: String,
+            description: String,
+            isRichContent: Bool = false
+        ) {
+            sectionData = [
+                .init(title: title, description: description, isRichContent: isRichContent)
+            ]
+        }
 
-        public init(title: String, description: String) {
-            self.title = title
-            self.description = description
+        public init(_ sectionData: [SectionData]) {
+            self.sectionData = sectionData
         }
 
+        public init(_ sectionData: SectionData?) {
+            self.sectionData = (sectionData == nil ? [] : [sectionData!])
+        }
+
+        @ViewBuilder
         public var body: some View {
-            VStack(spacing: 0) {
-                InstUI.Divider()
-
-                VStack(alignment: .leading, spacing: 4) {
-                    Text(title)
-                        .textStyle(.infoTitle)
-                    Text(description)
-                        .textStyle(.infoDescription)
+            if sectionData.isEmpty {
+                SwiftUI.EmptyView()
+            } else {
+                VStack(spacing: 0) {
+                    InstUI.Divider()
+
+                    VStack(alignment: .leading, spacing: 0) {
+                        ForEach(sectionData) { sectionData in
+                            VStack(alignment: .leading, spacing: 4) {
+                                Text(sectionData.title)
+                                    .textStyle(.infoTitle)
+
+                                if sectionData.isRichContent {
+                                    WebView(
+                                        html: sectionData.description,
+                                        features: [.disableDefaultBodyMargin],
```

Comment:
I think this results in a strange look in dark mode when theme is switched via Switch button.
The margins will stay dark, and only the content will change to light. I can live with it, just wanted to mention.
![Screenshot 2024-04-23 at 13 04 44](https://github.com/instructure/canvas-ios/assets/15140172/2bed1f15-0d85-4b17-9f2e-5d312df5ba6d)